apps/openmw/mwgui/statswatcher.cpp

```diff
@@ -0,0 +1,209 @@
+#include "statswatcher.hpp"
+
+#include "../mwbase/environment.hpp"
+#include "../mwbase/windowmanager.hpp"
+#include "../mwbase/world.hpp"
+
+#include "../mwmechanics/npcstats.hpp"
+#include "../mwmechanics/spellutil.hpp"
+
+#include "../mwworld/class.hpp"
+#include "../mwworld/esmstore.hpp"
+#include "../mwworld/inventorystore.hpp"
+
+namespace MWGui
+{
+    // mWatchedTimeToStartDrowning = -1 for correct drowning state check,
+    // if stats.getTimeToStartDrowning() == 0 already on game start
+    StatsWatcher::StatsWatcher()
+      : mWatchedLevel(-1), mWatchedTimeToStartDrowning(-1), mWatchedStatsEmpty(true)
+    {
+    }
+
+    void StatsWatcher::watchActor(const MWWorld::Ptr& ptr)
+    {
+        mWatched = ptr;
```

Comment:
Should this method set `mWatchedStatsEmpty` to true if ptr is empty? Otherwise update will not notify when other ptr is started to be watched until attribute change.

Reply:
Maybe. I just moved existing code.

Reply:
I think we can leave this as is. Unlikely that we will find some bug because of this.